### DISCUSSION TITLE
Testing defer results

### DISCUSSION THREAD

**Discussion by chillbram** [laravel/framework#52961] [Q&A] [answered]
Is there a clean way to test the results of [deferred functions](https://laravel.com/docs/11.x/helpers#deferred-functions)? Ideally, I would like to have the ability for `defer()` to simply run the closure in place, analogous to the way we can use the `sync` driver for queues.

**Comment by Jennoramcaj** [comment]
It’s not the best way to do it, but you can use this workaround:
```php
$this->get('/');
```

Since “defer” is executed **after an HTTP response**, you can simulate a simple HTTP request to ensure that defer gets executed.

**Comment by chillbram** [comment, marked as answer]
Solution is now available with #53340, which is available from v11.30.0 onwards thanks to @timacdonald!
